monkestation/code/modules/antagonists/borers/code/focus_datum.dm

```diff
@@ -0,0 +1,77 @@
+/datum/borer_focus
+	/// Name of the focus
+	var/name = ""
+	/// Cost of the focus
+	var/cost = 5
+	/// Traits to add/remove
+	var/list/traits = list()
+
+/// Effects to take when the focus is added
+/datum/borer_focus/proc/on_add(mob/living/carbon/human/host, mob/living/basic/cortical_borer/borer)
+	SHOULD_CALL_PARENT(TRUE)
+	for(var/trait in traits)
+		if(HAS_TRAIT(host, trait))
+			continue
+		ADD_TRAIT(host, trait, REF(borer))
```

Comment:
this is the reason trait sources exist, because some host traits might be temporary, and if they're present whenever the focus is added, the focus won't work
```suggestion
	for(var/trait in traits)
		ADD_TRAIT(host, trait, REF(borer))
```